lib/read.js

```diff
@@ -74,8 +74,48 @@ function read (req, res, next, parse, debug, options) {
 
   // read body
   debug('read body')
-  getBody(stream, opts, function (error, body) {
-    if (error) {
+  new Promise(function (resolve, reject) {
```

Comment:
Why does this need to be wrapped in a new promise? The raw-body module already returns a promise.

Reply:
Oh, ok :slightly_smiling_face: I didn't realize that. I presumed that if callback was used then it's a callback-only API. I'll change my PR to use the returned promise instead :+1: